src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -325,5 +328,61 @@ public virtual void EnableExtensions(bool enable = true)
             var rc = raw.sqlite3_enable_load_extension(_db, enable ? 1 : 0);
             SqliteException.ThrowExceptionForRC(rc, _db);
         }
+
+        internal (sqlite3_stmt stmt, string tail) GetPreparedStatement(string commandText)
+        {
+            var tail = commandText;
+            var rc = raw.sqlite3_prepare_v2(
+                _db,
+                tail,
+                out sqlite3_stmt stmt,
+                out tail);
+            SqliteException.ThrowExceptionForRC(rc, _db);
+
+            if (stmt.ptr != IntPtr.Zero)
+            {
+                _preparedStatements.Add(stmt);
+            }
+
+            return (stmt, tail);
+        }
+
+        internal void GetPreparedStatements(string commandText, ICollection<sqlite3_stmt> stmts)
+        {
+            var tail = commandText;
+            sqlite3_stmt stmt;
+            do
+            {
+                (stmt, tail) = GetPreparedStatement(tail);
+
+                // Statement was empty, white space, or a comment
+                if (stmt.ptr == IntPtr.Zero)
+                {
+                    if (!string.IsNullOrEmpty(tail))
+                    {
+                        continue;
+                    }
+
+                    break;
+                }
```

Comment:
Some of this code is duplicated in `ExecuteReader`. Can we share more of the implementation somehow?